src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -19,28 +22,36 @@
 
     private String mergeRemote;
     private String mergeTarget;
+    private String mergeSource;
     private String mergeStrategy;
     private MergeCommand.GitPluginFastForwardMode fastForwardMode;
-
     /**
      * @deprecated use the new constructor that allows to set the fast forward mode.
      */
     @Deprecated
-    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy) {
-        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
+    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeSource, String mergeStrategy) {
+        this(mergeRemote, mergeTarget, mergeSource, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
     }
-
     @DataBoundConstructor
-    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy,
+    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeSource, String mergeStrategy,
             MergeCommand.GitPluginFastForwardMode fastForwardMode) {
         this.mergeRemote = mergeRemote;
         this.mergeTarget = mergeTarget;
+        this.mergeSource = mergeSource;
```

Comment:
It would be better to use <code>hudson.Util</code> to fix nulls and to forget about empty strings